src/MassTransit/SystemTextJsonMessageSerializer.cs

```diff
@@ -21,7 +21,8 @@ public class SystemTextJsonMessageSerializer : IMessageSerializer
                 context.ContentType = JsonContentType;
 
                 var envelope = new JsonMessageEnvelope(context, context.Message, TypeMetadataCache<T>.MessageTypeNames);
-                AsyncUtil.RunSync(() => JsonSerializer.SerializeAsync<MessageEnvelope>(stream, envelope, SystemTextJsonConfiguration.Options));
+                JsonSerializer.SerializeAsync<MessageEnvelope>(stream, envelope, SystemTextJsonConfiguration.Options)
```

Comment:
why not just use Serialize(), there is no benefit to faking async here.

Reply:
You are right but Microsoft has didn't provide sync version of this method :)

Reply:
done - sync now.